### DISCUSSION TITLE
survey-likert question names in data

### DISCUSSION THREAD

**Discussion by ellamoeck** [jspsych/jsPsych#864] [Q&A] [answered]
Hi there,
I am using the survey-likert plugin and am trying to get the names of questions to save in the datafile, rather than Q0, Q1 etc. I followed the instructions and added names for each question, but when the data downloads it does not show the question names. An example of one of the question sets is below. Is there something else I need to add either here or to the survey-likert plugin script?
Thanks,
Ella

`var BSQ1 = {
	type: "survey-likert",
	preamble: '<p style="font-size:100%">We would like to know how you have been feeling about your appearance over the past four weeks. Please read each question and select the appropriate response. Please answer all questions.<br/><br/><b>Over the PAST FOUR WEEKS:</b></p>',
	questions: [
		{prompt: "1. Has feeling bored made you brood about your shape?", name: 'BSQ1bored_brood', labels: scale_1},
		{prompt: "2. Have you been so worried about your shape that you felt you ought to diet?", name: 'BSQ2worried_diet', labels: scale_1},
		{prompt: "3. Have you thought your thighs, hips or bottom are disproportionate to the rest of your body?", name: 'BSQ3disproportionate', labels: scale_1},
		{prompt: "4. Have you been afraid that your weight might increase and/or decrease too much?", name: 'BSQ4weight_fluct', labels: scale_1},
		{prompt: "5. Have you worried that your body is not toned enough?", name: 'BSQ5toned', labels: scale_1},
	],
	randomize_question_order: false,
};`

**Comment by legatrix** [comment]
Hi Ella,

I'm no expert so wait for a better answer, but just to say I quickly ran your code and the question names seem to be fine in the data output. Code:

```
<!DOCTYPE html>
<html>
	<head>
		<title>Likert Scale Test</title>
		<script src = "jsPsych-6.1.0/jspsych.js"></script>
		<script src = "jsPsych-6.1.0/plugins/jspsych-survey-likert.js"></script>
		<link href="jsPsych-6.1.0/css/jspsych.css" rel="stylesheet" type="text/css"></link>
	</head>
	<body></body>
	<script>

		var timeline = [];

		var scale_1 = [
  			"Strongly Disagree", 
  			"Disagree", 
  			"Neutral", 
  			"Agree", 
  			"Strongly Agree"
		];

		var BSQ1 = { type: "survey-likert",
			preamble: '<p style="font-size:100%">We would like to know how you have been feeling about your appearance over the past four weeks. Please read each question and select the appropriate response. Please answer all questions.<br/><br/><b>Over the PAST FOUR WEEKS:</b></p>',
			questions: [ {prompt: "1. Has feeling bored made you brood about your shape?",
					name: 'BSQ1bored_brood', labels: scale_1},
				{prompt: "2. Have you been so worried about your shape that you felt you ought to diet?",
					name: 'BSQ2worried_diet', labels: scale_1},
				{prompt: "3. Have you thought your thighs, hips or bottom are disproportionate to the rest of your body?",
					name: 'BSQ3disproportionate', labels: scale_1},
				{prompt: "4. Have you been afraid that your weight might increase and/or decrease too much?",
					name: 'BSQ4weight_fluct', labels: scale_1},
				{prompt: "5. Have you worried that your body is not toned enough?", name: 'BSQ5toned', labels: scale_1}, ],
			randomize_question_order: false, };

		timeline.push(BSQ1);

		jsPsych.init({
			timeline: timeline,
			on_finish: function(){
				jsPsych.data.displayData();
			}
		});

	</script>
</html>
```

Data:

```
[
	{
		"rt": 6650,
		"responses": "{\"BSQ1bored_brood\":2,\"BSQ2worried_diet\":2,\"BSQ3disproportionate\":2,\"BSQ4weight_fluct\":2,\"BSQ5toned\":2}",
		"question_order": "[0,1,2,3,4]",
		"trial_type": "survey-likert",
		"trial_index": 0,
		"time_elapsed": 6652,
		"internal_node_id": "0.0-0.0"
	}
]
```

Luke

**Reply by ellamoeck** [reply]
Thanks Luke! Good to know it wasn't my code causing the issue.

**Comment by jodeleeuw** [comment, marked as answer]
Hi Ella,

This feature is new as of version 6.1.0. Are you using an older version of the library?

**Reply by ellamoeck** [reply]
Ah! I needed to update just about all of the plugins due to adapting a code previously written in 2017.
Thanks Josh.
